scripts/esbuild/internal-platform-client.ts

```diff
@@ -0,0 +1,110 @@
+import type { BuildOptions as ESBuildOptions } from 'esbuild';
+import { replace } from 'esbuild-plugin-replace';
+import fs from 'fs-extra';
+import glob from 'glob';
+import { join } from 'path';
+
+import { getBanner } from '../utils/banner';
+import { BuildOptions, createReplaceData } from '../utils/options';
+import { writePkgJson } from '../utils/write-pkg-json';
+import { externalAlias, getBaseEsbuildOptions, getEsbuildAliases, getEsbuildExternalModules } from './util';
+
+/**
+ * Create objects containing ESbuild options for the two bundles which need to
+ * be written to `internal/client`. This also performs relevant side-effects,
+ * like clearing out the directory and writing a `package.json` script to disk.
+ *
+ * @param opts build options
+ * @returns an array of ESBuild option objects
+ */
+export async function getInternalClientBundle(opts: BuildOptions): Promise<ESBuildOptions[]> {
+  const inputClientDir = join(opts.srcDir, 'client');
+  const outputInternalClientDir = join(opts.output.internalDir, 'client');
+  const outputInternalClientPolyfillsDir = join(outputInternalClientDir, 'polyfills');
+
+  await fs.emptyDir(outputInternalClientDir);
+  await fs.emptyDir(outputInternalClientPolyfillsDir);
+
+  await copyPolyfills(opts, outputInternalClientPolyfillsDir);
+
+  // write @stencil/core/internal/client/package.json
+  writePkgJson(opts, outputInternalClientDir, {
+    name: '@stencil/core/internal/client',
+    description:
+      'Stencil internal client platform to be imported by the Stencil Compiler and internal runtime. Breaking changes can and will happen at any time.',
+    main: 'index.js',
+    sideEffects: false,
+  });
+
+  const internalClientAliases = getEsbuildAliases();
+  internalClientAliases['@platform'] = join(inputClientDir, 'index.ts');
+
+  const clientExternal = getEsbuildExternalModules(opts, opts.output.internalDir);
+
+  const internalClientBundle: ESBuildOptions = {
+    ...getBaseEsbuildOptions(),
+    entryPoints: [join(inputClientDir, 'index.ts')],
+    format: 'esm',
+    outfile: join(outputInternalClientDir, 'index.js'),
+    platform: 'node',
+    external: clientExternal,
+    alias: internalClientAliases,
+    banner: {
+      js: getBanner(opts, 'Stencil Client Platform'),
+    },
+    plugins: [
+      replace(createReplaceData(opts)),
+      externalAlias('@app-data', '@stencil/core/internal/app-data'),
+      externalAlias('@utils/shadow-css', './shadow-css.js'),
+      // we want to get the esm, not the cjs
```

Comment:
Just because we're outputting esm here. I think without re-pointing this import to the esm version esbuild was giving me an error saying 'nothing is exported from this module' (because it was trying to import cjs as esm). I'll add a little explanation there